### ISSUE TITLE
ö,ä,å not working on Motorola Milestone

### ISSUE THREAD

**Issue by eis** [irssiconnectbot/irssiconnectbot#7]
How to reproduce the problem?
Try typing ä,ö or å. In Milestone, that should be done by pressing 'a' or 'o' for a long time and then selecting the desired character, but this does not work with ICB, instead just repeating 'a'.

What is the excpected output?
Special characters that can be typed with long presses would be usable.

What version of the product are you using?
1.7.1

**Comment by voneiden** [comment]
I had a similar problem in the stock HTC Legend sense keyboard. When you long press the key 'a', the keyboard typed the 'a' already. Then the menu for choosing accents comes up, you select 'ä', and it would delete the 'a' and replace it with 'ä'. However I assume the terminal did not recognize the delete and the ä became lost somewhere.

Technically I would imagine it's a bug in the keyboard software, but as I've switched to Cyanogenmod a year ago, I can't recall the specifics of this problem. Of course, ConnectBot could feature a compatibility layer to fix this problem.

**Comment by eis** [comment]
Menu for choosing accents never comes up for my Milestone, irrelevant of how long I wait. I'm not sure if there is any way for me to produce the characters except copy & paste...

**Comment by voneiden** [comment]
From market you can find a bunch of replacement keyboards.  "Scandinavian Keyboard" and "MultiLing Keyboard" are good (free) examples of keyboards that have a proper QWERTY layout which includes ä,ö,å buttons. Makes typing a lot more enjoyable. Dictionaries are available to both too from the market.

**Comment by eis** [comment]
Yeah, I probably can. Point of this bug would be however to get irssi connectbot to work like the other apps, respecting my current keyboard configuration. I don't want to install additional keyboards to my phone to get this one app working.

**Comment by voneiden** [comment]
Ah, so it's actually a physical keyboard you're talking here about?

FYI irssi connectbot also has the native input box. If you tap at the bottom of the SSH window, 5 buttons should become visible for a short period of time. Tapping on the 2nd from right (the one that looks like an empty rectangle) will open a native input box. I'm pretty sure your keyboard works as expected on that one. Of course it's not very convenient.

Also, I'm pretty sure this is a connectbot issue, not ICB issue. And unfortunately for you, it seems to be a design principle, quoting straight from the connectbot wiki: "_Since ConnectBot doesn't use any of the normal TextView widgets, Android's IME structure isn't designed to directly support it._". 

I assume the Milestone's keyboard's accent functionality depends on TextView widgets.

**Comment by eis** [comment]
Yes, a physical keyboard. You're right, it works! Thank you for this. It is not very convenient, but definitely better than not having the keys at all.
